ebean-core/src/main/java/io/ebeaninternal/server/core/DatabasePlatformFactory.java

```diff
@@ -159,7 +174,7 @@ private DatabasePlatform byDatabaseMeta(DatabaseMetaData metaData, Connection co
     } else if (dbProductName.contains("sqlite")) {
       return new SQLitePlatform();
     } else if (dbProductName.contains("db2")) {
-      return new DB2Platform();
+      throw new IllegalArgumentException("For DB2 please explicitly choose either db2legacy/db2luw/db2zos/db2fori platform. Refer to issue #2514 for details");
```

Comment:
@rbygrave Should we make some autodetection here?

Reply:
I am happy for it to be explicit.  Maybe go with this for now and see later on if auto-detection is worthwhile?